test/e2e/apimachinery/namespace.go

```diff
@@ -282,4 +286,74 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
 		framework.ExpectEqual(namespace.ObjectMeta.Labels["testLabel"], "testValue", "namespace not patched")
 	})
 
+	ginkgo.It("should apply changes to a namespace status", func() {
+		nsResource := v1.SchemeGroupVersion.WithResource("namespaces")
+		dc := f.DynamicClient
+		ns := f.Namespace.Name
+
+		ginkgo.By("Read namespace status")
+
+		unstruct, err := dc.Resource(nsResource).Get(context.TODO(), ns, metav1.GetOptions{}, "status")
+		framework.ExpectNoError(err, "failed to fetch NamespaceStatus %s", ns)
+		nsStatus, err := unstructuredToNamespace(unstruct)
+		framework.ExpectNoError(err, "Getting the status of the namespace %s", ns)
+		framework.ExpectEqual(nsStatus.Status.Phase, v1.NamespaceActive, "The phase returned was %v", nsStatus.Status.Phase)
+		framework.Logf("Status: %#v", nsStatus.Status)
+
+		ginkgo.By("Patch namespace status")
+		patchStatus, err := json.Marshal(v1.NamespaceCondition{
+			Type:    "StatusPatch",
+			Status:  v1.ConditionTrue,
+			Reason:  "E2E",
+			Message: "Patched by an e2e test",
+		})
+		framework.ExpectNoError(err, "failed to marshal unstructured response")
+
+		payload := "{\"status\": {\"conditions\": [" + string(patchStatus) + "]}}"
```

Comment:
Is there a reason to use this construct vs what's in job status (https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/job.go#L488) or vice versa?

Wasn't clear to me whether this was part of the test (subtly)

Reply:
What sort of issues?  Surprising to me on namespace, so makes me curious.

Reply:
I think it's mostly my mental model of the problem. Also, as `namespace.status` only has `conditions   <[]Object>` or `phase        <string>` my preference was to test changes with a set of `conditions`. With a fresh set of eyes on the problem I've got an local copy of the test using `f.Clientset` but still patching/checking for a new set of `conditions`.

Reply:
I changed the test around to follow the existing tests/style as much as I can.